components/ILIAS/LegalDocuments/classes/Conductor.php

```diff
@@ -99,6 +88,23 @@ public function logoutText(): string
         return null === $logout_text ? '' : $this->container->ui()->renderer()->render($logout_text());
     }
 
+    public function logoutTarget(LogoutTarget $target): LogoutTarget
+    {
+        $user_withdrew_legal_docs = false;
```

Comment:
You don't need this variable (`$user_withdrew_legal_docs`). You can just return the `$target` in the `catch` block and else return `ForcedLogoutTarget`. You don't need the `if` in the `ForcedLogoutTarget`.

I also noticed that we can make this even simpler: As we don't use `$id` we can just check for existence with `$this->container->http()->wrapper()->query()->has('withdraw_consent');` instead of the try catch block.